deeplearning4j-core/src/test/java/org/deeplearning4j/nn/layers/custom/testclasses/CustomLayer.java

```diff
@@ -79,4 +80,9 @@ public void setNIn(InputType inputType, boolean override) {
     public InputPreProcessor getPreProcessorForInputType(InputType inputType) {
         return null;
     }
+
+    @Override
+    public LayerMemoryReport getMemoryReport(InputType inputType) {
+        throw new UnsupportedOperationException();
+    }
```

Comment:
Autodiff should give us the memory estimation functionality also, essentially for free for each layer using it.
Currently it's all manual here, which isn't ideal for sure. I do like the idea of virtual execution though - that would be a much better solution long-term.
We can do another pass in the future for all that - for now, this'll have to do.